src/EditorFeatures/CSharpTest/CodeActions/InlineTemporary/InlineTemporaryTests.cs

```diff
@@ -3736,6 +3736,28 @@ public void M()
     {
         var s2 = string.Replace($""hello"", ""world"");
     }
+}", parameters: new TestParameters(parseOptions: new CSharpParseOptions(LanguageVersion.CSharp7)));
```

Comment:
Change in overload resolution affects the logic in `IsUnnecessaryCast` in error cases. I pinged @gafter to confirm whether this is intended.

```C#
        [Fact]
        public void Test73()
        {
            var comp = CreateCompilation(@"class C { void M(string s) { string.Replace(s, s); } }", parseOptions: TestOptions.Regular7_3);

            var tree = comp.SyntaxTrees.Single();
            var model = comp.GetSemanticModel(tree, ignoreAccessibility: false);
            var node = tree.GetRoot().DescendantNodes().OfType<InvocationExpressionSyntax>().Single();
            Assert.Equal("string.Replace(s, s)", node.ToString());

            var symbolInfo = model.GetSymbolInfo(node);
            Assert.Equal(new[] { "System.String System.String.Replace(System.Char oldChar, System.Char newChar)",
                "System.String System.String.Replace(System.String oldValue, System.String newValue)" }, 
                symbolInfo.CandidateSymbols.Select(s => s.ToTestDisplayString()));
        }

        [Fact]
        public void Test7()
        {
            var comp = CreateCompilation(@"class C { void M(string s) { string.Replace(s, s); } }", parseOptions: TestOptions.Regular7);

            var tree = comp.SyntaxTrees.Single();
            var model = comp.GetSemanticModel(tree, ignoreAccessibility: false);
            var node = tree.GetRoot().DescendantNodes().OfType<InvocationExpressionSyntax>().Single();
            Assert.Equal("string.Replace(s, s)", node.ToString());

            var symbolInfo = model.GetSymbolInfo(node);
            Assert.Empty(symbolInfo.CandidateSymbols);
        }
```

Reply:
Thanks, I can see how this would make sense after the bestest betterness/candidate filtering work.

Reply:
I would have expected the reverse (more candidates listed in 7 than in 7.3).

Reply:
Yes but my understanding is that originally the test used 7.0 (I think IDE tests use `default` by default, not `latest`)

Reply:
Or maybe I misunderstood... what is the output of the 2 tests you posted above with 7.0 and 7.3?

Reply:
Ok sorry I misread the tests (thought 7.0 was first). That's definitely interesting, unless it's actually a Symbol (therefore not inside CandidateSymbols) in 7.0. Isn't that the case? Then this seems like a good change.

Reply:
Neal confirmed this behavior is expected: "A method that is called that way is not even a candidate in 7.3. Since no method is a candidate, we list some failed candidates in SymbolInfo.CandidateSymbols."